File: C++/NucleusLib/pnSceneObject/plSimulationInterface.cpp
#include "plSimulationInterface.h"

plSimulationInterface::plSimulationInterface() : Unknown(0) {
    Physical = new plKey();
}

plSimulationInterface::~plSimulationInterface() {
    delete Physical;
}

void plSimulationInterface::read(hsStream *S) {
    plObjInterface::read(S);

    SimFlags.read(S);
    Unknown = S->readInt();
    Physical->readRef(S);
}

void plSimulationInterface::write(hsStream *S) {
    plObjInterface::write(S);

    SimFlags.write(S);
    S->writeInt(Unknown);
    Physical->writeRef(S);
}

File: Java/Plasma/NucleusLib/pnSceneObject/plSimulationInterface.java
package Plasma.NucleusLib.pnSceneObject;

import java.io.IOException;
import Plasma.CoreLib.hsStream;
import Plasma.CoreLib.hsBitVector;
import Plasma.NucleusLib.pnKeyedObject.*;

public class plSimulationInterface extends plObjInterface {
    protected hsBitVector SimFlags = new hsBitVector();
    protected int Unknown = 0;
    protected plKey Physical = new plKey();

    public void read(hsStream S) throws IOException {
        super.read(S);

        SimFlags.read(S);
        Unknown = S.readInt();
        Physical.readRef(S);
    }

    public void write(hsStream S) throws IOException {
        super.write(S);

        SimFlags.write(S);
        S.writeInt(Unknown);
        Physical.writeRef(S);
    }
}

